src/Migrations/HSLMigration.php

```diff
@@ -0,0 +1,646 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST\Migrations;
+use namespace Facebook\HHAST;
+use namespace HH\Lib\{C, Str, Vec, Math};
+
+use type Facebook\HHAST\{
+  EditableNode,
+  FunctionCallExpression,
+  NameToken,
+  EditableList,
+  BinaryExpression,
+  LiteralExpression,
+  BooleanLiteralToken,
+  NullLiteralToken,
+  NamespaceGroupUseDeclaration,
+  NamespaceUseDeclaration,
+  QualifiedName,
+  Script,
+  INamespaceUseDeclaration,
+  NamespaceUseClause,
+  NamespaceToken,
+  ListItem,
+  PrefixUnaryExpression,
+  MinusToken,
+  DecimalLiteralToken,
+  LiteralExpression,
+  ExpressionStatement,
+  OctalLiteralToken,
+  CommaToken,
+  WhiteSpace,
+};
+
+use function Facebook\HHAST\__Private\find_type_for_node;
+
+enum HSL_NAMESPACE: string as string {
```

Comment:
is the `as string` required?

Reply:
Nope! I removed it and put `coerce()` in a few other places.